File: learnAPI/3.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

url <- "https://randomuser.me/api/"

response <- GET(url)

if (http_status(response)$category == "Success") {
    json_content <- content(response, "text")
    data <- fromJSON(json_content)

    df <- as.data.frame(data)
    print(head(df))
    cat('\n\n\n')

    df <- data.frame(
        Gender = data$results$gender,
        Title = data$results$name$title,
        `Full Name` = paste(data$results$name$first, data$results$name$last, sep = ' '),
        city = data$results$location$city,
        state = data$results$location$state,
        country = data$results$location$country
    )
    print(df)
}

File: learnAPI/3.py
import pandas as pd
import requests

url = 'https://randomuser.me/api/'

response = requests.get(url)

if response.status_code == 200:
    data = response.json()
    
    data1 = data['results'][0]
    print(data1)
    print()
    data = data['results'][0]
    print(data)
    
    df = pd.DataFrame({
    "Gender": [data["gender"]],
    "Title": [data["name"]["title"]],
    "First Name": [data["name"]["first"]],
    "Last Name": [data["name"]["last"]],
    'Full Name': [data['name']['title'] + ' ' + data["name"]["first"] + ' ' + data["name"]["last"]],
    "Street Number": [data["location"]["street"]["number"]],
    "Street Name": [data["location"]["street"]["name"]],
    "City": [data["location"]["city"]],
    "State":[data["location"]["state"]],
    "Country": [data["location"]["country"]],
    })
    print(df)
    
else:
    print("Failed to retrieve data:", response.status_code)